docs/examples/oceanmodelling.py

```diff
@@ -0,0 +1,509 @@
+        ax.set(xlabel="Training iteration", ylabel="Negative marginal log likelihood")
+
+    return opt_posterior
+
+
+opt_velocity_posterior = optimise_mll(velocity_posterior, dataset_train)
+
+
+# %% [markdown]
+# ### Comparison
+# We next obtain the latent distribution of the GP of $g$ at $\mathbf{x}_{0,i}$, then extract its mean and standard at the test locations, $\mathbf{F}_{\text{latent}}(\mathbf{x}_{0,i})$, as well as the standard deviation (we will use it at the very end).
+
+
+# %%
+def latent_distribution(opt_posterior, pos_3d, dataset_train):
+    latent = opt_posterior.predict(pos_3d, train_data=dataset_train)
+    latent_mean = latent.mean()
+    latent_std = latent.stddev()
+    return latent_mean, latent_std
+
+
+# extract latent mean and std of g, redistribute into vectors to model F
+velocity_mean, velocity_std = latent_distribution(
+    opt_velocity_posterior, dataset_ground_truth.X, dataset_train
+)
+
+dataset_latent_velocity = dataset_3d(pos_test, velocity_mean)
+
+
+# %% [markdown]
+# We now replot the ground truth (testing data) $D_0$, the predicted latent vector field $\mathbf{F}_{\text{latent}}(\mathbf{x_i})$, and a heatmap of the residuals at each location $\mathbf{R}(\mathbf{x}_i) = \mathbf{y}_{0,i} - \mathbf{F}_{\text{latent}}(\mathbf{x}_i) $, as well as $\left|\left|\mathbf{R}(\mathbf{x}_i)\right|\right|$.
+
+
+# %%
+# Residuals between ground truth and estimate
+
+
+def plot_vector_field(ax, dataset, **kwargs):
+    ax.quiver(
+        dataset.X[::2][:, 0],
+        dataset.X[::2][:, 1],
+        dataset.y[::2],
+        dataset.y[1::2],
+        **kwargs,
+    )
+
+
+def prepare_ax(ax, X, Y, title, **kwargs):
+    ax.set(
+        xlim=[X.min() - 0.1, X.max() + 0.1],
+        ylim=[Y.min() + 0.1, Y.max() + 0.1],
+        aspect="equal",
+        title=title,
+        ylabel="latitude",
+        **kwargs,
+    )
+
+
+def residuals(dataset_latent, dataset_ground_truth):
+    return jnp.sqrt(
+        (dataset_latent.y[::2] - dataset_ground_truth.y[::2]) ** 2
+        + (dataset_latent.y[1::2] - dataset_ground_truth.y[1::2]) ** 2
+    )
+
+
+def plot_fields(
+    dataset_ground_truth, dataset_trajectory, dataset_latent, shape=shape, scale=10
+):
+    X = dataset_ground_truth.X[:, 0][::2]
+    Y = dataset_ground_truth.X[:, 1][::2]
+    # make figure
+    fig, ax = plt.subplots(1, 3, figsize=(12.0, 2.7), sharey=True)
+
+    # ground truth
+    plot_vector_field(
+        ax[0],
+        dataset_ground_truth,
+        color=colors[0],
+        label="Ocean Current",
+        angles="xy",
+        scale=scale,
+    )
+    plot_vector_field(
+        ax[0],
+        dataset_trajectory,
+        color=colors[1],
+        label="Drifter",
+        angles="xy",
+        scale=scale,
+    )
+    prepare_ax(ax[0], X, Y, "Ground Truth")
+
+    # Latent estimate of vector field F
+    plot_vector_field(ax[1], dataset_latent, color=colors[0], angles="xy", scale=scale)
+    plot_vector_field(
```

Comment:
These plots look nice but the left hand plot is mixing `Longitude` on the x-axis.

Also legend for the blue and red colours over labs with the central plots `Longitude` label - need a bit a spacing between.